### ISSUE TITLE
Why need workspace when fuse BN and ReLu?

### ISSUE THREAD

**Issue by CaoZhongZ** [intel/mkl-dnn#583]
# Summary
We observed BatchNorm fused with ReLu (training) have worse performance than not to.

# Expected behavior
Better performance

I noticed that when fused BN with ReLu in training it required an extra space to write to. It doesn't have to be. Activation sometimes have a property that the gradient can be calculated solely from "y" without the need of "x". That's the basic theory a ReLu could fused into another operator even in training.

**Comment by CaoZhongZ** [comment]
hmmm, makes sense. We tested ncsp batch norm and the data from different scenarios gave us contradictory story.
